src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/SemanticTokenTestBase.cs

```diff
@@ -114,8 +145,77 @@ protected string GetBaselineFileContents(string baselineFileName)
             csharpTokens = result?.Data;
         }
 
-        var csharpResponse = new ProvideSemanticTokensResponse(tokens: csharpTokens, hostDocumentSyncVersion);
-        return csharpResponse;
+        return new ProvideSemanticTokensResponse(tokens: csharpTokens, hostDocumentSyncVersion);
+    }
+
+    // Duplicated from SemanticTokens.cs
+    private int[] StitchSemanticTokenResponsesTogether(int[][] responseData)
```

Comment:
Should expose the real method through a test accessor, otherwise we we could inadvertantly break the logic in it, and the tests would happily continue to pass.

Reply:
A few observations I noticed as part of this PR:

1. The `StitchSemanticTokenResponsesTogether` logic is under `Microsoft.VisualStudio.LanguageServerClient.Razor` project and not accessible in this the `Microsoft.AspNetCore.Razor.LanguageServer.Test` project which contains `SemanticTokenTestBase.cs` above and the tests containing generated baseline files.

2. I found it confusing that `RazorCustomMessageTarget` class is partially implemented in a file with name `SemanticTokens.cs` where it contains `StitchSemanticTokenResponsesTogether` method. I am intentionally leaving the file name unchanged, because I assumed the intention was for it to be in line with the `SemanticTokenTestBase.cs` filename here.

3. @davidwengier I share your concern.

- I can look into adding a new test in `RazorCustomMessageTargetTest.cs` because over there the `StitchSemanticTokenResponsesTogether` method can be accessible for testing.

Reply:
Alternatively, consider moving the stitch logic over to the language server, so custom message target only does the bare minimum of code to pass messages and responses around.

That would also make it easier for VS Code if we want to have this new behaviour there: any logic in custom message target has to be re-implemented in type script in the C# extension, so the less logic, the better.

Reply:
> I left the filename unchanged, because I assumed the intention was for it to be in line with the SemanticTokenTestBase.cs filename here.

The intention is to split the `RazorCustomMessageTarget` class up into multiple files, each concerned with a different feature, so its easier to find and work on. Unfortunately technical limitations mean it has to be one giant class, so I thought this would be easier to work with. It was only split up like this 3 weeks ago though: https://github.com/dotnet/razor/pull/9080